docs/java-rest/low-level/configuration.asciidoc

```diff
@@ -64,21 +64,75 @@ include-tagged::{doc-tests}/RestClientDocumentation.java[rest-client-config-disa
+
+Configuring the client to authenticate with an elasticsearch access token can be accomplished by setting the relevant HTTP request header.
+If the client makes requests on behalf of a single user only, you can set the necessary `Authorization` header as a default header as shown
+in the following example
+
+["source","java",subs="attributes,callouts,macros"]
+--------------------------------------------------
+include-tagged::{doc-tests}/RestClientDocumentation.java[rest-client-auth-bearer-token]
+--------------------------------------------------
+
+==== Elasticsearch API keys
+
+Configuring the client to authenticate with an elasticsearch API Key can be also accomplished by setting the relevant HTTP request header.
+If the client makes requests on behalf of a single user only, you can set the necessary `Authorization` header as a default header as shown
+in the following example
+
+["source","java",subs="attributes,callouts,macros"]
+--------------------------------------------------
+include-tagged::{doc-tests}/RestClientDocumentation.java[rest-client-auth-api-key]
+--------------------------------------------------
+
 === Encrypted communication
 
-Encrypted communication can also be configured through the
+Encrypted communication using TLS can also be configured through the
 `HttpClientConfigCallback`. The
 https://hc.apache.org/httpcomponents-asyncclient-dev/httpasyncclient/apidocs/org/apache/http/impl/nio/client/HttpAsyncClientBuilder.html[`org.apache.http.impl.nio.client.HttpAsyncClientBuilder`]
  received as an argument exposes multiple methods to configure encrypted
  communication: `setSSLContext`, `setSSLSessionStrategy` and
  `setConnectionManager`, in order of precedence from the least important.
- The following is an example:
+
+When accessing an elasticsearch cluster that is setup for TLS on the HTTP layer, the client needs to trust the certificate that
+elasticsearch is using.
+ The following is an example of setting up the client to trust the CA that has signed the certificate that elasticsearch is using, when
+ that CA certificate is available in a PKCS#12 keystore:
 
 ["source","java",subs="attributes,callouts,macros"]
 --------------------------------------------------
 include-tagged::{doc-tests}/RestClientDocumentation.java[rest-client-config-encrypted-communication]
 --------------------------------------------------
 
+The following is an example of setting up the client to trust the CA that has signed the certificate that elasticsearch is using, when
+that CA certificate is available as a PEM encoded file.
+
+["source","java",subs="attributes,callouts,macros"]
+--------------------------------------------------
+include-tagged::{doc-tests}/RestClientDocumentation.java[rest-client-config-trust-ca-pem]
+--------------------------------------------------
+
+When elasticsearch is configured to require client TLS authentication, i.e. when a PKI realm is configured, the client needs to provide
```

Comment:
> When elasticsearch is configured to require client TLS authentication, i.e. when a PKI realm is configured ...

The use of "i.e." here implies that this is the only situation where ES requires TLS authentication, which seems unlikely to me.  Should it be "e.g." or "for example" instead?

Reply:
A PKI realm is actually the most common case to require TLS Client authentication on the HTTP layer. Otherwise the client authenticates with some other means ( credentials, tokens, api keys ) and not with a client certificate. I

Reply:
Unless it's the *only* case, I'd still recommend "e.g." instead of "i.e.".

Reply:
It is the _only_ documented case , but a user might elect to enable it and just disregard the authentication information ( by not specifying a PKI realm ) I guess, so I'll apply the suggestion. 

By the way, thanks for that , I did not know that distinction and was using e.g and i.e. interchangeably for years :heart:

Reply:
Words are fun i.e. crazy-making :)